protocol/flatbuffers/data_feed.fbs

```diff
@@ -0,0 +1,124 @@
+
+
+include "datatypes/hardware_info.fbs";
+include "datatypes/datatypes.fbs";
+include "datatypes/imu.fbs";
+
+namespace slimevr_protocol.data_feed;
+
+/// Requests for a single `DeviceStatus` to be sent. This is helpful when getting
+/// initial info about the device.
+table PollDeviceStatus {
+    mask: DeviceStatusMask;
+}
+
+
+/// Requests for the other party to send `data_feeds`.
+/// For example, GUI requests for position data to be sent from server.
+//
+// TODO: Should we add a `DataFeedCancel`? I think its fine to just kill the connection
+//   to do that. We don't need *that* much control.
+table DataFeedRequest {
+    data_feeds: [DataFeedConfig];
+}
+
+/// Notifies the other party that `data_feeds` will be sent.
+/// For example, feeder app notifies server that controller data will be sent.
+//
+// TODO: Perhaps its enough to just start sending the data feed info? Do we really
+//   need to notify the other end that we are going to do it? Or perhaps its also useful
+//   to respond to a `DataFeedRequest` and tell them which feeds will *actually* be
+//   supported. I'm not sold on the necessity of this either way, I'd rather just
+//   let the application handle the case of data being missing, and respond to incoming
+//   data feeds automatically.
+table DataFeedNotify {
+    data_feeds: [DataFeedConfig];
+}
+
+
+/// All information related to the configuration of a data feed. This may be sent
+/// as part of a `DataFeedRequest` or a `DataFeedNotify`.
+table DataFeedConfig {
+    /// Minimum delay in milliseconds between new data updates.
+    min_interval: uint16;
+    data_mask: DeviceStatusMask;
+
+    // Note: I removed control over only sending on update, etc, because it seemed
+    // unecessarily complicated to implement
+}
+
+/// A mask of values to be reported in subsequent DeviceStatus. Values set to `false`
+/// or `null` will not reported. By default, all fields are false/null.
+///
+/// If you set a value to `true`, it is not guaranteed that the sender actually has
+/// such a value to send. In this case, they will probably send `null`, and the receiver
+/// has the choice to disconnect due to missing data.
+//
+// TODO: Instead of the receiver guessing if the sender has the needed data, should
+//   the sender instead first send a `DeviceStatusMask` back, to indicate what subset
+//   of data they can actually provide? Downside is the network code might be harder
+//   to implement.
+table DeviceStatusMask {
+    custom_name: bool;
+    role: bool;
+    is_synthetic: bool;
+    firmware_info: slimevr_protocol.datatypes.hardware_info.FirmwareInfoMask;
+    imu_data: slimevr_protocol.datatypes.imu.ImuDataMask;
+    firmware_status: slimevr_protocol.datatypes.hardware_info.FirmwareStatusMask;
+}
+
+/// All possible info related to a tracked device. Tracked devices are for example
+/// a SlimeVR tracker, a vive tracker, or a simulated Vive tracker. We may add more
+/// tracked devices in the future.
+table DeviceStatus {
```

Comment:
It shoud be called Tracker Status and not Device.

Because a device is kinda = to an MPU but a tracker is an IMU / Sensor.
Each trackers has a role. Devices have multiple trackers that each have a role

Reply:
Resolved by having `DeviceStatus` and `TrackerStatus`, and making `DataFeedUpdate` have a list of `DeviceStatus` and a list of synthetic `TrackerStatus`. `DeviceStatus` also has `TrackerStatus`